src/bfx_tools/picard.ml

```diff
@@ -150,6 +150,55 @@ let mark_duplicates
       on_failure_activate (Remove.file ~run_with metrics_path);
     ]
 
+let reorder_sam
+    ~(run_with: Machine.t) ?mem_param
+    ?reference_build ~input_bam output_bam_path
+  =
+  let open KEDSL in
+  let picard_jar = Machine.get_tool run_with Machine.Tool.Default.picard in
+  let tmp_dir =
+    Filename.chop_extension output_bam_path ^ ".tmpdir" in
+  let input_bam_path = input_bam#product#path in
+  let name =
+    sprintf "picard-reorder_sam-%s" Filename.(basename input_bam#product#path) in
+  let product = transform_bam input_bam#product output_bam_path in
```

Comment:
In the case where `reference_build` is `Some nrb`, the new bam should carry the new `nrb` as its ref-build I think.